### PR TITLE
Maintenance - python 3.6+, pydicom 1.0+

### PR DESCRIPTION
**Pull request by jond01**
Hi, before starting to work on #18 I wanted to update some of dicom-numpy's code.
Sorry for the absence of a preceding discussion, we can decide here what to take or not from this PR.

This PR includes:
- Transition to **python 3.6+** only, as currently, it is the oldest maintained python version.
   The resulting changes are:
  - Add tests for python 3.8 and 3.9, drop support for python 2.7 and 3.5
  - F-strings instead of `.format`
  - Use `unittest.mock`, `math.isclose` and `open` built-ins
- Pydicom 1.0+: update requirements, tests and docs
- Add `__version__` to the package's `__init__.py`, so that users can check `dicom_numpy.__version__`.
  `setup.py` imports this definition and uses it: `version=__version__`
- Misc.:
  - Update the year in the license
  - Fix pydicom links
  - Change rows and columns interpretation in the tests (does not affect their integrity, just an internal definition)

### CODE REVIEW

File: dicom_numpy/combine_slices.py [hunk 1]
```diff
@@ -158,7 +158,7 @@ def _validate_slices_form_uniform_grid(slice_datasets):
         'SamplesPerPixel',
         'PixelSpacing',
         'PixelRepresentation',
-        'BitsAllocated',
+        'BitsAllocated'
```
Comment (johndgiese) [inline dicom_numpy/combine_slices.py:161]: I think we should keep this comma, according to PEP8's recommendation:

https://www.python.org/dev/peps/pep-0008/#when-to-use-trailing-commas

File: dicom_numpy/combine_slices.py [hunk 2]
```diff
@@ -180,19 +180,19 @@ def _validate_image_orientation(image_orientation):
     row_cosine, column_cosine, slice_cosine = _extract_cosines(image_orientation)
 
     if not _almost_zero(np.dot(row_cosine, column_cosine), 1e-4):
-        raise DicomImportException("Non-orthogonal direction cosines: {}, {}".format(row_cosine, column_cosine))
```
Comment (johndgiese) [inline dicom_numpy/combine_slices.py:183]: 👍

File: dicom_numpy/zip_archive.py [hunk 3]
```diff
@@ -36,21 +33,21 @@ def dicom_datasets_from_zip(zip_file):
         entry_pseudo_file = zip_file.open(entry)
 
         # the pseudo file does not support `seek`, which is required by
-        # dicom's lazy loading mechanism; use temporary files to get around this;
+        # pydicom's lazy loading mechanism; use temporary files to get around this;
```
Comment (johndgiese) [inline dicom_numpy/zip_archive.py:36]: 👍

File: docs/source/index.rst [hunk 4]
```diff
@@ -4,7 +4,7 @@ DICOM-Numpy
 
 This python module provides a set of utilities for extracting data contained in
 DICOM files into Numpy ndarrays.  It is a higher-level library that builds on the excellent lower-level `pydicom
-<http://pydicom.readthedocs.io/en/stable/>`_ library.
+<https://pydicom.github.io/pydicom/>`_ library.
```
Comment (johndgiese) [inline docs/source/index.rst:7]: 👍

File: tests/conftest.py [hunk 5]
```diff
@@ -31,18 +31,22 @@ def __init__(self, pixel_array, slice_position, row_cosine=None, column_cosine=N
 
         shape = pixel_array.shape
         if len(shape) == 2:
-            num_columns, num_rows = shape
+            num_rows, num_columns = shape
             samples_per_pixel = 1
         else:
-            num_columns, num_rows, samples_per_pixel = shape
+            num_rows, num_columns, samples_per_pixel = shape
+            # TODO: add
```
Comment (johndgiese) [inline tests/conftest.py:38]: If you don't mind, could you add a little more context for this TODO?

**Review by johndgiese — COMMENTED** [review]
@jond01 very nice cleanup PR---I learned a few things looking at it!

I left two small comments. I was a bit hesitant about dropping Python 2.7 and pydicom < 1.0 support, but after some thought, I feel it is for the best. Hopefully we don't have any 2.7 users who get upset about this.

Thank you for taking the time to do such a nice PR.

File: dicom_numpy/combine_slices.py (hunk 1 above)
Comment (jond01) [reply dicom_numpy/combine_slices.py:161]: Right, interesting

File: tests/conftest.py (hunk 5 above)
Comment (jond01) [reply tests/conftest.py]: Sure, I added information in the comment.
This TODO is just about supporting a rare (nonexistent maybe?) but allowed situation of planar configuration = 1, which means that the RGB channels are along the first axis of `pixel_array`.
Usually, planar configuration = 0, and the channels are last.
Anyway, a dicom series with RGB slices is itself a pretty uncommon situation, though dicom-numpy supports it.

**Pull request merged**